indico/htdocs/js/indico/jquery/categorynavigator.js

```diff
@@ -95,66 +111,82 @@
 
             // Visible search field
             self.$searchInput = $('<input>', {
-                class: 'js-search-category js-typeahead',
+                class: 'js-search-category',
                 type: 'search',
                 name: 'q',
                 placeholder: $T.gettext("Search")
             }).attr('autocomplete', 'off');
+            self.element.prepend(self.$searchInput);
+
+            self.$searchInput.on('input', _.debounce(function(evt) {
+                var $this = $(this);
+                var inputValue = $this.val().trim();
 
-            // Insert elements in DOM
-            var $typeaheadForm = $($.parseHTML(self._typeaheadTemplate)[0]);
-            $typeaheadForm.find('.typeahead__query').append(self.$searchInput);
-            self.element.prepend($typeaheadForm);
-
-            // Typeahead init
-            self.$searchInput.typeahead({
-                dynamic: true,
-                filter: false,
-                source: {
-                    ajax: {
+                if (inputValue.length > 2) {
```

Comment:
Why not return if it's `< 3` instead of having a huge block of code inside the if body and nothing outside?